qutebrowser/commands/runners.py

```diff
@@ -285,6 +286,12 @@ def run(self, text, count=None):
             else:
                 result.cmd.run(self._win_id, args)
 
+            if (result.cmdline[0] != 'repeat-command' and
+                    result.cmd.mode_allowed(usertypes.KeyMode.normal)):
```

Comment:
Why does this only record normal commands? Wouldn't it make more sense to record recent commands *per mode*, so you could bind/use `.` in other modes as well?